### ISSUE TITLE
training using the train_triplet file

### ISSUE THREAD

**Issue by Yasmineldemery** [polimi-ispl/icpr2020dfdc#58]
Hi,
I wanted to train ur model and train the efficientnetb4ST so I started with the triplet file just as you mentioned in the script however I keep getting this message, is there any way to fix this?
I' m using googlecolab for training 

![image](https://user-images.githubusercontent.com/64609398/124910364-cc21ae00-dfeb-11eb-9ecc-27ed492f0243.png)

**Comment by nicobonne** [comment]
Hey @Yasmineldemery. You are running out of GPU memory. You should lower the `--batch` parameter of the training script, which is 12 by default, until you manage to fit the available memory on your system.

**Comment by Yasmineldemery** [comment]
@nicobonne Thanks a lot, I tried it with 16 and it worked, I have another question though, after training and testing the model, I saved the results as a csv file to view them and I dont quite understand the results, Can u help me understand such results.

I though the scores would be from 0 to 1 however i get results ranging between negative 0.somthing values and up tp 3.something

![image](https://user-images.githubusercontent.com/64609398/124936138-92aa6c00-e006-11eb-83c6-5a371df8602a.png)

**Comment by nicobonne** [comment]
Are you sure it is working with 16? I mean, it's a bit strange it exceeds the memory with 12 and works with 16 :)

Regarding the scores: those are pre-sigmoid scores. To obtain the scores in 0-1 range you need to pass them to torch sigmoid function or `expit` from scipy. This is an intended behaviour as we compute the sigmoid in the loss function during training, so the pure predictions of the network are pre-sigmoid.

**Comment by Yasmineldemery** [comment]
@nicobonne yess as i was using 32 at first,
 Also pls forgive me as I'm still new to this field, can u tell mehow to pass the scores to a sigmoid function? 
where is this done? in the test_model file or where?
and also how cause i really dont know :)
Thanks in advance

**Comment by nicobonne** [comment]
OK, I thought you were using the default value of 12 :)

Applying the sigmoid is as simple as doing `torch.sigmoid(yourscores)`. Have a look to the penultimate cell of the [image prediction notebook](https://github.com/polimi-ispl/icpr2020dfdc/blob/master/notebook/Image%20prediction.ipynb) for a reference.

**Comment by Yasmineldemery** [comment]
@nicobonne 
is this correct?
![image](https://user-images.githubusercontent.com/64609398/124939491-6f34f080-e009-11eb-9d84-ae9f49ad28f2.png)

**Comment by nicobonne** [comment]
It should be fine, but I suggest you to use [`scipry.special.expit()`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.special.expit.html) if you are treating your scores as numpy array and not torch tensors.

**Comment by Yasmineldemery** [comment]
do u mean replacing this
torch.sigmoid(df['score'])

with this

scipy.special.expit(df['score']) ?

**Comment by Yasmineldemery** [comment]
okay i tried it however the results were the same, i got results outside the 0-1 range :/ 

Is there a way to use this "torch.sigmoid(df['score'])" without resulting in an error?

**Comment by Yasmineldemery** [comment]
@nicobonne  I tried changing the scores to Tensor and it worked.
Many thanks for your help :)
